app/src/behaviors/behavior_sensor_rotate.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2022 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#define DT_DRV_COMPAT zmk_behavior_sensor_rotate
+
+#include <device.h>
+#include <drivers/behavior.h>
+#include <logging/log.h>
+
+#include <drivers/sensor.h>
+#include <zmk/event_manager.h>
+#include <zmk/events/keycode_state_changed.h>
+#include <zmk/behavior_queue.h>
+#include <zmk/keymap.h>
+
+LOG_MODULE_DECLARE(zmk, CONFIG_ZMK_LOG_LEVEL);
+
+struct behavior_sensor_rotate_config {
+    struct zmk_behavior_binding cw_binding;
+    struct zmk_behavior_binding ccw_binding;
+};
+
+struct behavior_sensor_rotate_data {
+    struct zmk_behavior_binding *triggered_binding;
+};
+
+static int on_sensor_binding_triggered(struct zmk_behavior_binding *binding,
+                                       const struct device *sensor, int64_t timestamp) {
+    const struct device *dev = device_get_binding(binding->behavior_dev);
+    const struct behavior_sensor_rotate_config *cfg = dev->config;
+    struct behavior_sensor_rotate_data *data = dev->data;
+
+    struct sensor_value value;
+    int err;
+
+    err = sensor_channel_get(sensor, SENSOR_CHAN_ROTATION, &value);
+
+    if (err) {
+        LOG_WRN("Failed to get sensor rotation value: %d", err);
+        return err;
+    }
+
+    switch (value.val1) {
+    case 1:
+        data->triggered_binding = (struct zmk_behavior_binding *)&cfg->cw_binding;
+        break;
+    case -1:
+        data->triggered_binding = (struct zmk_behavior_binding *)&cfg->ccw_binding;
+        break;
+    default:
+        return -ENOTSUP;
+    }
+
+    LOG_DBG("Sensor binding: %s", log_strdup(binding->behavior_dev));
+
+    zmk_behavior_queue_add(0, *data->triggered_binding, true, 0);
+    zmk_behavior_queue_add(0, *data->triggered_binding, false, 5);
```

Comment:
The 5 here is arbitrary, and not great. Maybe we should make this an optional value in the DT, that defaults to 5 to keep the existing functionality unchanged?